planetiler-core/src/main/java/com/onthegomap/planetiler/util/CommonStringEncoder.java

```diff
@@ -1,54 +1,74 @@
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.AtomicInteger;
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * A utility for compressing up to 250 commonly-used strings (i.e. layer name, tag attributes) into a single byte.
+ * A utility for compressing commonly-used strings (i.e. layer name, tag attributes).
  */
 @ThreadSafe
 public class CommonStringEncoder {
 
-  private final ConcurrentMap<String, Byte> stringToId = new ConcurrentHashMap<>(255);
-  private final String[] idToString = new String[255];
-  private final AtomicInteger layerId = new AtomicInteger(0);
+  private static final int MAX_STRINGS = 100_000;
+
+  private final Map<String, Integer> stringToId = new ConcurrentHashMap<>(MAX_STRINGS);
+  private final String[] idToString = new String[MAX_STRINGS];
+  private final AtomicInteger stringId = new AtomicInteger(0);
 
   /**
    * Returns the string for {@code id}.
    *
    * @throws IllegalArgumentException if there is no value for {@code id}.
    */
-  public String decode(byte id) {
-    String str = idToString[id & 0xff];
+  public String decode(int id) {
+    String str = idToString[id];
     if (str == null) {
       throw new IllegalArgumentException("No string for " + id);
     }
     return str;
   }
 
   /**
-   * Returns a byte value to each unique string passed in.
+   * Returns a int value to each unique string passed in.
    *
    * @param string the string to store
-   * @return a byte that can be converted back to a string by {@link #decode(byte)}.
+   * @return an int that can be converted back to a string by {@link #decode(int)}.
    * @throws IllegalArgumentException if called for too many values
    */
-  public byte encode(String string) {
+  public int encode(String string) {
     // optimization to avoid more expensive computeIfAbsent call for the majority case when concurrent hash map already
     // contains the value.
-    Byte result = stringToId.get(string);
+    Integer result = stringToId.get(string);
     if (result == null) {
       result = stringToId.computeIfAbsent(string, s -> {
-        int id = layerId.getAndIncrement();
-        if (id > 250) {
-          throw new IllegalArgumentException("Too many string keys when inserting " + string);
+        int id = stringId.getAndIncrement();
+        if (id >= MAX_STRINGS) {
+          throw new IllegalArgumentException("Too many strings");
         }
         idToString[id] = string;
-        return (byte) id;
+        return id;
       });
     }
     return result;
   }
+
+  /**
+   * Variant of CommonStringEncoder based on byte rather than int for string indexing.
+   */
+  public static class AsByte {
+    private final CommonStringEncoder encoder = new CommonStringEncoder();
+
+    public String decode(byte id) {
+      return encoder.decode(id & 0xff);
+    }
+
+    public byte encode(String string) {
+      if (encoder.stringId.get() > 255) {
```

Comment:
If we add a constructor parameter to CommonStringEncoder with the max number of allowed strings (could default to 100k), then we could get rid of this check here.